demo/setup.sh

```diff
@@ -1,10 +1,13 @@
 #!/bin/bash
 
 set -e
+dir="$(dirname "${BASH_SOURCE[0]}")"
+: ${PYTHON:=python}
 
-if [[ ! -f ~/envs/plop-demo/bin/activate ]]; then
-    virtualenv -p python2.7 --no-site-packages ~/envs/plop-demo
+if [[ ! -f "$dir/envs/plop-demo/bin/activate" ]]; then
+    "$PYTHON" -mvirtualenv --no-site-packages "$dir/envs/plop-demo" || \
+        "$PYTHON" -mvenv "$dir/envs/plop-demo"
```

Comment:
The original intent was to prefer venv (even though you are correct I got it wrong), because it is in the standard library of recent Pythons.  Likewise, pip is (more or less) part of the stdlib since the inclusion of ensurepip (but calling `python -mensurepip` first does seem a bit too pedantic).
Either I can fix it or you can mark virtualenv as a requirement for the demos, I don't really care.